Main.cpp

```diff
@@ -252,6 +252,34 @@ void _LoadingSplashProgressReport(std::string reportString, bool newItem = false
 
 void LoadingSplashProgressReport(std::string reportString, bool newItem = false)
 {
+	if(g_LogToCli)
+	{
+		if (newItem)
+		{
+			std::cout << std::endl;
+		}
+		// overwrite current line
+		std::cout << "\r";
+		size_t startPos = 0;
+		// convert characters to unicode
+		// just make sure to really overwrite all old output
+		std::string unicoded = reportString + "          ";
+		// also uses terminal coloring
+		std::string to = "\033[1;32m✓\033[0;0m";
+		while ((startPos = unicoded.find(-42, startPos)) != std::string::npos)
```

Comment:
Can you put some comment to explain the stuff in the while loops? I guess it's getting replacing any *s with a green checkmark and any +s with a yellow circle, but comments that explain that would make it a lot easier. Also the variable name to is super undescriptive, at least to me. I'd rather you expand that to mean something when it's read.